python/metatensor-operations/metatensor/operations/sort.py

```diff
@@ -4,6 +4,79 @@
 from ._classes import Labels, TensorBlock, TensorMap, torch_jit_is_scripting
 
 
+def _sort_single_gradient_block(
+    block: TensorBlock,
+    gradient_block: TensorBlock,
+    axes: List[str],
+    descending: bool,
+) -> TensorBlock:
+    """
+    Sorts a single TensorBlock without the user input checking and sorting of gradients
+    """
+
+    sample_names = gradient_block.samples.names
+    sample_values = gradient_block.samples.values
+
+    component_names: List[List[str]] = []
+    components_values = []
+    for component in gradient_block.components:
+        component_names.append(component.names)
+        components_values.append(component.values)
+
+    property_names = gradient_block.properties.names
+    properties_values = gradient_block.properties.values
+
+    values = gradient_block.values
+    if "samples" in axes:
```

Comment:
just to be clear, you don't re-sort the "sample" column but update it according to the values samples sorting. Does this re-sort the other columns in the samples?

Reply:
I updated the in-code doc a little bit

Reply:
sounds good! I had a hard time following some of this